src/engine/authentication.md

```diff
@@ -35,10 +35,24 @@ The HMAC algorithm implies that several CLs will be able to use the same key, an
 
 The `EL` and `CL` clients **MUST** accept a cli/config parameter: `jwt-secret`, which designates a file containing the hex-encoded 256 bit secret key to be used for verifying/generating JWT tokens.
 
-If such a parameter is not given, the client **SHOULD** generate such a token, valid for the duration of the execution, and store the hex-encoded secret as a `jwt.hex` file on the filesystem.  This file can then be used to provision the counterpart client.
-
 If such a parameter _is_ given, but the file cannot be read, or does not contain a hex-encoded key of `256` bits, the client should treat this as an error: either abort the startup, or show error and continue without exposing the authenticated port.
 
+If such a parameter is not given, the client **MUST** attempt to read the secret from the default paths defined below, in the order they are listed.  If a secret is found, but the file cannot be read, does not contain a hex-encoded key of `256` bit, or is rejected by the other client, the client should continue searching. The CL **MAY** search other locations, such as default EL data directories.
```

Comment:
> the client should continue searching

and ultimately abort if no valid file is found?
marginal worry it implies "search forever"

Reply:
Updated to more clearly state when an abort should occur.